### PR TITLE
sonar issues

### PR DESCRIPTION
**Pull request by tandraschko**
sonar issues

@melloware can you please review?

### CODE REVIEW

File: primefaces/src/main/java/org/primefaces/component/datatable/DataTableRenderer.java [hunk 1]
```diff
@@ -980,11 +980,16 @@ public void encodeTbody(FacesContext context, DataTable table, boolean dataOnly,
...
-            int virtualScrollRowCount = (table.getScrollRows() * 2);
-            rowCountToRender = Math.min(virtualScrollRowCount, rowCount);
+            rowCountToRender = Math.min(table.getScrollRows() * 2, rowCount);
+        }
+        else if (table.isLiveScroll()) {
+            rowCountToRender = table.getScrollRows() + table.getScrollOffset();
+        }
+        else {
+            rowCountToRender = rows == 0 ? rowCount : rows;
         }
```
Comment (melloware) [inline primefaces/src/main/java/org/primefaces/component/datatable/DataTableRenderer.java:993]: I think this logic needs to be 

```java
int rowCountToRender;

if (rows == 0) {
    if (table.isVirtualScroll()) {
        rowCountToRender = Math.min(table.getScrollRows() * 2, rowCount);
    }
    else if (table.isLiveScroll()) {
        rowCountToRender = table.getScrollRows() + table.getScrollOffset();
    }
    else {
        rowCountToRender = rowCount;
    }
} else {
    rowCountToRender = rows;
}
```

**Review by melloware — CHANGES_REQUESTED** [review]

File: primefaces/src/main/java/org/primefaces/component/datatable/DataTableRenderer.java (hunk 1 above)
Comment (tandraschko) [reply primefaces/src/main/java/org/primefaces/component/datatable/DataTableRenderer.java:993]: the virtualScoll if was outside before
wait 1 min, maybe this is more clear

Comment (melloware) [reply primefaces/src/main/java/org/primefaces/component/datatable/DataTableRenderer.java:993]: Yeah it looks like they were specifically looking for 0 rows before attempting the other checks?

Comment (melloware) [reply primefaces/src/main/java/org/primefaces/component/datatable/DataTableRenderer.java:993]: Sorry I updated my code to say "virtualScroll" i had LiveScroll twice.

Comment (tandraschko) [reply primefaces/src/main/java/org/primefaces/component/datatable/DataTableRenderer.java:993]: not sure TBH
nows its 100% the old logic again

**Review by melloware — APPROVED** [review]

**Pull request merged**